### ISSUE TITLE
Add a suffix field to the bottle DSL

### ISSUE THREAD

**Issue by osembedded** [Homebrew/homebrew#43091]
I have a custom tap hosted on a github.com with some formulas and bottles. The problem I have is that github adds the query parameter ?raw=text to the bottles URL if you want to download it as binary. Is there a way I can specify a suffix parameter in the DSL?

Note: It makes more sense for me to host both the formula and the bottles on one github repo.

For example:
bottle do
  root_url "http://mikemcquaid.com"
  prefix "/opt/homebrew"
  <b>suffix "?raw=text"</b>
  cellar "/opt/homebrew/Cellar"
  revision 4
  sha256 "4921af80137af9cc3d38fd17c9120da882448a090b0a8a3a19af3199b415bfca" => :yosemite
  sha256 "c71db15326ee9196cd98602e38d0b7fb2b818cdd48eede4ee8eb827d809e09ba" => :mavericks
  sha256 "85cc828a96735bdafcf29eb6291ca91bac846579bcef7308536e0c875d6c81d7" => :mountain_lion
end

**Comment by mikemcquaid** [comment]
We don't support a suffix. Also, you shouldn't host binary bottles on GitHub. Instead consider using e.g. Bintray.com.

**Comment by osembedded** [comment]
Just to clarify, this is a private enterprise github account and mainly used for host tools. So, I don't think hosting a few bottles on it would be a big issue. Having the bottles alongside the formulas makes it easier to keep them in sync and maintain.

I guess the other option is to host both the formula and the bottles on an internal web server but I lose some of the advantages of git like versioning, commit history etc.

**Comment by mikemcquaid** [comment]
If it's an Enterprise account: :cool:. I'd rather not extend the DSL for this case though but if I see more requests in future is I'll consider it.

**Comment by mistydemeo** [comment]
I wonder if the [releases](https://help.github.com/articles/about-releases/) feature might be more suitable for you to host your binaries than hosting them in the repository?

**Comment by osembedded** [comment]
@mistydemeo - Thanks for the tip. I tried what you suggested and it almost worked. The issue I have is that when brew tries to download the file from the releases link, it cannot authenticate with the enterprise github account and instead ends up downloading the login prompt webpage instead of the actual release. It makes me wonder... even if we implemented the url suffix feature, we may have the same authentication issue with github from the command line.
